### DISCUSSION TITLE
How could I get vcpkg's cryptopp built with asan make target (address sanitizer)

### DISCUSSION THREAD

**Discussion by rturrado** [microsoft/vcpkg#25068] [Q&A] [answered]
I have a CMake project that uses vcpkg as a submodule to install some dependencies, cryptopp between them.

Now, in order to build this project with ASAN (address sanitizer, in debug mode), it seems I need to compile all dependencies with ASAN; otherwise I get a link error such as `error LNK2038: mismatch detected for 'annotate_vector': value '0' doesn't match value '1' in` (this seems to be the case for Visual Studio 2022 17.2 and beyond; check [here](https://docs.microsoft.com/en-us/answers/questions/864574/enabling-address-sanitizer-results-in-error-lnk203.html) and [here](https://docs.microsoft.com/en-us/cpp/sanitizers/error-container-overflow?view=msvc-170)).

I've tried defining adding the `-fsanitize=address` to `CMAKE_CPP_FLAGS` or `CRYPTOPP_COMPILE_OPTIONS` before the first `project()` in any CMakeLists.txt file, thinking that way those vars would be passed to vcpkg before building cryptopp; but that's not working. 

What are my options to get, under certain build configurations, a vcpkg's cryptopp build with ASAN?

**Comment by Osyotr** [comment, marked as answer]
Use overlay triplet and set `VCPKG_CXX_FLAGS` in it.
https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#vcpkg_cxx_flags

**Reply by rturrado** [reply]
Many thanks for the quick answer @Osyotr.
